src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +109,49 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    protected boolean isRequestROEUpdate(HttpServletRequest request, String transactionId) {
+        final Map<String, List<String>> privileges = getERICTokenPermissions(request);
+        String companyInScope;
+        if (privileges.containsKey(COMPANY_NUMBER_KEY)) {
+            companyInScope = privileges.get(COMPANY_NUMBER_KEY).get(0);
+            if (StringUtils.isNotEmpty(companyInScope) && companyInScope.startsWith("OE")) {
+                String companyInTransaction = getCompanyInTransaction(request, transactionId);
+                if (StringUtils.isNotBlank(companyInTransaction)) {
+                    return companyInScope.equalsIgnoreCase(companyInTransaction);
```

Comment:
I would write a specific log error message if the company numbers don't match. It should never happen but if it does, it should be investigated by support.